plugins/outputs/gw8/gw8.go

```diff
@@ -0,0 +1,447 @@
+	cacheKeyTracerToken = "cacheKeyTraceToken"
+	tracerCache         = cache.New(-1, -1)
+)
+
+var (
+	sampleConfig = `
+  ## HTTP endpoint for your groundwork instance.
+  endpoint = ""
+
+  ## Agent uuid for Groundwork API Server
+  agent_id = ""
+
+  ## Groundwork application type
+  app_type = ""
+
+  ## Username to access Groundwork API
+  username = ""
+
+  ## Password to use in pair with username
+  password = ""
+
+  ## Default display name for the host with services(metrics)
+  default_host = "default_telegraf"
+
+  ## Default service state [default - "SERVICE_OK"]
+  default_service_state = "SERVICE_OK"
+
+  ## The name of the tag that contains the hostname [default - "host"]
+  resource_tag = "host"
+`
+)
+
+type GW8 struct {
+	Server              string `toml:"groundwork_endpoint"`
+	AgentID             string `toml:"agent_id"`
+	AppType             string `toml:"app_type"`
+	Username            string `toml:"username"`
+	Password            string `toml:"password"`
+	DefaultHost         string `toml:"default_host"`
+	DefaultServiceState string `toml:"default_service_state"`
+	ResourceTag         string `toml:"resource_tag"`
+	authToken           string
+}
+
+func (g *GW8) SampleConfig() string {
+	return sampleConfig
+}
+
+func (g *GW8) Connect() error {
+	if g.Server == "" {
+		return errors.New("no 'server' provided")
+	}
+
+	byteToken, err := login(g.Server+loginURL, g.Username, g.Password)
+	if err == nil {
+		g.authToken = string(byteToken)
+	}
+
+	return err
+}
+
+func (g *GW8) Close() error {
+	formValues := map[string]string{
+		"gwos-app-name":  "gw8",
+		"gwos-api-token": g.authToken,
+	}
+
+	headers := map[string]string{
+		"Content-Type": "application/x-www-form-urlencoded",
+	}
+
+	_, _, err := addons.SendRequest(http.MethodPost, g.Server+logoutURL, headers, formValues, nil)
+
+	return err
+}
+
+func (g *GW8) Write(metrics []telegraf.Metric) error {
+	resourceToServicesMap := make(map[string][]addons.DynamicMonitoredService)
+	for _, metric := range metrics {
+		resource, service := parseMetric(g.DefaultHost, g.DefaultServiceState, g.ResourceTag, metric)
+		resourceToServicesMap[resource] = append(resourceToServicesMap[resource], service)
+	}
+
+	var resources []addons.DynamicMonitoredResource
+	for resourceName, services := range resourceToServicesMap {
+		resources = append(resources, addons.DynamicMonitoredResource{
+			Name:          resourceName,
+			Type:          addons.Host,
+			Status:        addons.HostUp,
+			LastCheckTime: addons.Now(),
+			Services:      services,
+		})
+	}
+
+	requestJSON, err := json.Marshal(addons.DynamicResourcesWithServicesRequest{
+		Context: &addons.TracerContext{
+			AppType:    g.AppType,
+			AgentID:    g.AgentID,
+			TraceToken: makeTracerToken(),
+			TimeStamp:  addons.Now(),
+			Version:    addons.ModelVersion,
+		},
+		Resources: resources,
+		Groups:    nil,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	headers := map[string]string{
+		"GWOS-APP-NAME":  "gw8",
```

Comment:
I would rename the app to 'telegraf' since 'gw8' does not make any sense in the GroundWork logs.
```suggestion
		"GWOS-APP-NAME":  "telegraf",
```